chore(releasing): Prepare 0.20.0 release

jszwedko opened the pull request:
Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

JeanMertz commented on website/cue/reference/releases/0.20.0.cue:
```diff
@@ -0,0 +1,649 @@
+				- `stdin`
+				- `vector`
+				"""
+			pr_numbers: []
```
```suggestion
			pr_numbers: [10861, 10879, 10932, 11017, 10880, 11130, 10890, 10893, 11018, 11035, 11038, 11081, 11120, 11121, 11123, 11126, 11059]
```

jszwedko replied:
Thanks 😄

JeanMertz commented on website/cue/reference/releases/0.20.0.cue:
```diff
@@ -0,0 +1,649 @@
+			pr_numbers: [11042]
+		},
+		{
+			type: "fix"
```
```suggestion
			type: "enhancement"
```

JeanMertz commented on website/cue/reference/releases/0.20.0.cue:
```diff
@@ -0,0 +1,649 @@
+				This release includes a new opt-in disk buffer implementation that we hope will provide users with
+				faster, more consistent, and lower resource usage buffer performance. We encourage you to opt-in during
+				this beta period and [give us feedback](/community/). See [the beta disk buffer highlight
+				article](/highlights/2022-02-08-disk-buffer-v2-beta) for more details including how to opt-in.
```
As a potential improvement to release notes, we could pull out links to highlight articles into a structured element in the object.

We don't (necessarily) need to use it to render them in the description blurb, but it will help us in the future to link highlight articles back to their originating release note and/or linked PRs.

JeanMertz commented on website/cue/reference/releases/0.20.0.cue:
```diff
@@ -0,0 +1,649 @@
+				this beta period and [give us feedback](/community/). See [the beta disk buffer highlight
+				article](/highlights/2022-02-08-disk-buffer-v2-beta) for more details including how to opt-in.
+				"""
+			pr_numbers: []
```
I am uncertain if this _has_ to be a PR number, but this one seems to be good to link back to, as it includes links to all the PRs involved in getting this over the finish line: #9476.

pablosichert commented on distribution/kubernetes/vector-agent/rbac.yaml:
```diff
@@ -6,6 +6,11 @@ apiVersion: rbac.authorization.k8s.io/v1
+    app.kubernetes.io/name: vector
+    app.kubernetes.io/instance: vector
+    app.kubernetes.io/component: Agent
+    app.kubernetes.io/version: "0.19.0-distroless-libc"
```
Without knowing the context, just checking if `0.19.0` is intended here?

spencergilbert replied:
It ought to be 0.20.0... or maybe the install instructions have the user override it regardless to avoid a chicken & egg situation between these and the helm charts.

jszwedko replied:
Yeah, it is since these files are generated from the `helm-charts` which are still using 0.19.0. It's a bit of a chicken-and-egg situation with new vector releases. After the 0.20 release, I'll update the helm-charts and regenerate these assets.

spencergilbert replied:
Technically it's fine given the usage instructions we have - which do override the image so users won't get the wrong version... but :awkward:

jszwedko replied:
Hmm, yeah, that's true, we could have the generation script update to the new version before the images are published.

spencergilbert commented on website/cue/reference/releases/0.20.0.cue:
```diff
@@ -0,0 +1,649 @@
+				[Vector's component
+				specification](\(urls.specs_instrumentation)).
+
+				With this release we have instrumented the following sources with these new metrics:
```
And did we actually deprecate them?

jszwedko replied:
Yeah, I did notice that too. I think we can just have one release where they are all marked as deprecated rather than documenting their deprecation per-component.

jszwedko commented on website/cue/reference/releases/0.20.0.cue:
```diff
@@ -0,0 +1,649 @@
+			scopes: ["elasticsearch sink"]
+			description: """
+				Several deprecated `elasticsearch` options have been removed as described in the [upgrade
+				guide](highlights/2022-02-08-0-20-0-upgrade-guide#deprecate-v1). details.
```
👍 thanks for catching. We don't usually note documentation-only changes in the release notes so I'll just drop this.

jszwedko commented on website/cue/reference/releases/0.20.0.cue:
```diff
@@ -0,0 +1,649 @@
+			description: """
+				The `vector` source and `vector` sink now default to version 2 of the protocl for inter-Vector
+				communication. See the [upgrade guide](highlights/2022-02-08-0-20-0-upgrade-guide#deprecate-v1) for more
+				details.
```
It is, but we typically haven't documented how to skip versions. When users are upgrading Vector it is assumed they are upgrading one point release at a time. Later this will be one major-version at a time when we hit 1.0 and point releases won't have backwards incompatible changes.

jszwedko commented:
@JeanMertz thanks for the close review! I think I addressed all of the feedback in [a68c44d](https://github.com/vectordotdev/vector/pull/11164/commits/a68c44d763a1a44cb860f809d21beb24affe15a4)

JeanMertz reviewed (approved):
🚀
